packages/delegate/src/finalizeGatewayRequest.ts

```diff
@@ -547,7 +566,7 @@ function finalizeSelectionSet(
             return null;
           }
           const namedType = getNamedType(type);
-          if (!schema.getType(namedType.name) == null) {
+          if (schema.getType(namedType.name) == null) {
```

Comment:
is this change related? can you maybe elaborate? 
I'm asking because there might be a simpler way to flip the condition? 😆

Reply:
This condition was already incorrectly written. It should be like if the visited type is not found, return null.